src/viking.js

```diff
@@ -1,11 +1,87 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+    attack() {
+        return this.strength;
+    }
+    receiveDamage(damage) {
+        this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`
+        } else return `${this.name} has died in act of combat`;
+    }
+
+    battleCry() {
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`
+        } else return `A Saxon has died in combat`;
+    }
+}
+
 
 // War
-class War {}
+class War {
+    vikingArmy = [];
+    saxonArmy = [];
+    addViking(viking) {
+        this.vikingArmy.push(viking);
+    };
+    addSaxon(saxon) {
+        this.saxonArmy.push(saxon);
+    };
+    vikingAttack(){
+        const i = Math.floor(Math.random() * this.saxonArmy.length);
+        const randomSaxon = this.saxonArmy[i];
+        const j = Math.floor(Math.random() * this.vikingArmy.length);
+        const randomViking = this.vikingArmy[j];
+        const endOfVikingAttack = randomSaxon.receiveDamage(randomViking.attack());
+        endOfVikingAttack;        
+        if (randomSaxon.health <= 0) {
+            this.saxonArmy.splice(i,1);
+        }
+        return endOfVikingAttack;
+    };
+    saxonAttack(){
+        const i = Math.floor(Math.random() * this.saxonArmy.length);
+        const randomSaxon = this.saxonArmy[i];
+        const j = Math.floor(Math.random() * this.vikingArmy.length);
+        const randomViking = this.vikingArmy[j];
+        const endOfSaxonAttack = randomViking.receiveDamage(randomSaxon.attack());
+        endOfSaxonAttack;        
+        if (randomViking.health <= 0) {
+            this.vikingArmy.splice(i,1);
+        }
+        return endOfSaxonAttack;
+    };
+    showStatus(){
+        if (this.saxonArmy.length === 0) {
+            return "Vikings have won the war of the century!"
+        }
+        else if (this.vikingArmy.length === 0) {
```

Comment:
you can drop the elses here because you return from each of the if blocks anyways. 
this is called "early" return pattern.